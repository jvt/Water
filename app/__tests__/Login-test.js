jest.disableAutomock();

import {
    TextInput
} from 'react-native';
import Button from 'apsl-react-native-button';
import React from 'react';
import Login from '../login';
import Renderer from 'react-test-renderer';

describe('Login', () => {
    test('renders correctly', ()  => {
        const login = Renderer.create(
            <Login />
        ).toJSON();
        expect(login).toMatchSnapshot();
    });
});

