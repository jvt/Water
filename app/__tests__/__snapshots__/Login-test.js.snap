exports[`CancelButton renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Return
    </Text>
  </View>
</View>
`;

exports[`Login renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }>
        <Image
          source={1}
          style={
            Object {
              "height": 175,
              "width": 80,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }>
        Login
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Username"
        returnKeyType="next"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 70,
            "padding": 10,
          }
        } />
      <TextInput
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "padding": 10,
          }
        } />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(65, 163, 221, 1)",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 50,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Login
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Return
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View />
</View>
`;

exports[`LoginButton renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View
    style={
      Array [
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#0C42FD",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }>
      Login
    </Text>
  </View>
</View>
`;

exports[`PasswordField renders correctly 1`] = `
<TextInput
  style={Object {}} />
`;

exports[`UsernameField renders correctly 1`] = `
<TextInput
  style={Object {}} />
`;
